src/Microsoft.AspNetCore.Authentication.OpenIdConnect/Events/IOpenIdConnectEvents.cs

```diff
@@ -38,17 +28,22 @@ public interface IOpenIdConnectEvents : IRemoteAuthenticationEvents
         /// <summary>
         /// Invoked before redirecting to the identity provider to authenticate.
         /// </summary>
-        Task RedirectToAuthenticationEndpoint(RedirectContext context);
+        Task RedirectToIdentityProvider(RedirectContext context);
 
         /// <summary>
         /// Invoked before redirecting to the identity provider to sign out.
         /// </summary>
-        Task RedirectToEndSessionEndpoint(RedirectContext context);
+        Task RedirectToIdentityProviderForSignOut(RedirectContext context);
```

Comment:
@PinpointTownes it's OpenID Connect-specific, but it is not meant to be usable only by people who are familiar with OpenID Connect. Basic extensibility should be accessible to people who understand the high level pattern but aren't familiar with the trappings of the specific protocol. As the customizations get more granular, I agree that requiring familiarity is inevitable: but every developer understand what sign out is, without the need to know what the session endpoint is or how it works.